crates/astria-sequencer-relayer/src/relayer/mod.rs

```diff
@@ -167,11 +172,13 @@ impl Relayer {
             select!(
                 biased;
 
-                () = self.shutdown_token.cancelled() => {
+                () = self.relayer_shutdown_token.cancelled() => {
                     info!("received shutdown signal");
                     break Ok("shutdown signal received");
                 }
 
+                _ = &mut submitter_task => break Err(eyre!("Celestia submission task returned")),
```

Comment:
This is an unfortunate side-effect of changing to use child cancellation tokens - the submitter task (child) is no longer able to cancel the relayer task (parent).  Since the submitter task can exit with error during init, we need to poll for that here or else we hang in this `select!` loop.

This also means we need to `fuse` the submitter task and outside the loop we need to check whether it's already terminated or not before awaiting it, or else we'd hang there.

There are other potential approaches to address that, but I couldn't see anything cleaner than this (except for my still-preferred approach of a single cancellation token for all tasks rather than this hierarchy).